linkerd/meshtls/boring/src/creds/verify.rs

```diff
@@ -0,0 +1,78 @@
+use boring::x509::X509;
+use linkerd_dns_name as dns;
+use linkerd_identity as id;
+use std::io;
+
+pub(crate) fn verify_id(cert: &X509, id: &id::Id) -> io::Result<()> {
+    for san in cert.subject_alt_names().into_iter().flatten() {
+        if let Some(n) = san.dnsname() {
+            if let Ok(name) = n.parse::<dns::Name>() {
+                if *name == id.to_str() {
+                    return Ok(());
+                }
+            }
+            tracing::warn!("DNS SAN name {} could not be parsed", n)
```

Comment:
if we don't `return` because the name doesn't equal the identity, we will also fall through and print "DNS SAN name could not be parsed", even though that's not actually true in that case. i think this code should probably be more like this:

```suggestion
        match n.parse::<dns::Name>() {
            Ok(name) if *name == id.to_str() => return Ok(()),
            Ok(_) => {},
            Err(error) => tracing::warn!(%error, "DNS SAN name {} could not be parsed", n),
        }
```